### PR TITLE
FRO-40: spell simulator

### PR DESCRIPTION
**Pull request by tomeq93**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
     "lib": "lib"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "cast:gnosis": "sh scripts/fork-and-cast-spell.sh src/proposals/20230927/SparkGnosis_20230927.sol:SparkGnosis_20230927"
```
Comment (krzkaczor) [inline package.json]: i think we don't need to add it here.

File: scripts/fork-and-cast-spell.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+set -e
+
+trap "kill 0" EXIT
+
+spell_path_or_address=$1
+# default params for gnosis 
+spell_executor=${2-0xc4218C1127cB24a0D6c1e7D25dc34e10f2625f5A}
+anvil_fork_url=${3:-https://rpc.ankr.com/gnosis}
+anvil_fork_block_number=${4:-30031699}
+anvil_priv_key=${5-0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80}
```
Comment (krzkaczor) [inline scripts/fork-and-cast-spell.sh:10]: i think this default is the only one that makes sense to leave.

File: scripts/fork-and-cast-spell.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -e
+
+trap "kill 0" EXIT
+
+spell_path_or_address=$1
+# default params for gnosis 
+spell_executor=${2-0xc4218C1127cB24a0D6c1e7D25dc34e10f2625f5A}
```
Comment (krzkaczor) [inline scripts/fork-and-cast-spell.sh]: lets get rid of these default parameters. Can we raise an error instead if they are missing?

File: package.json (hunk 1 above)
Comment (krzkaczor) [reply package.json]: Let's create a readme section instead describing how to use the script

**Review by krzkaczor — CHANGES_REQUESTED** [review]

File: scripts/fork-and-cast-spell.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -e
+
+trap "kill 0" EXIT
+
+spell_path_or_address=$1
+spell_executor=$2
+anvil_fork_url=$3
+anvil_fork_block_number=$4
```
Comment (krzkaczor) [inline scripts/fork-and-cast-spell.sh:9]: last thing: can we make this optional? ie. default value should be latest block

**Review by krzkaczor — CHANGES_REQUESTED** [review]

File: scripts/fork-and-cast-spell.sh (hunk 4 above)
Comment (tomeq93) [reply scripts/fork-and-cast-spell.sh:9]: done - if not provided we skip --fork-block-number part

**Comment by tomeq93** [comment]
closed in https://github.com/marsfoundation/spark-spells/pull/19

**Pull request closed without merging**